plugin/pulpcore/plugin/stages/declarative_version.py

```diff
@@ -82,13 +81,8 @@ def __init__(self, first_stage, repository, sync_mode='mirror'):
                  :class:`~pulpcore.plugin.stages.DeclarativeVersion stream`, and does not remove any
                  pre-existing units in the :class:`~pulpcore.plugin.models.RepositoryVersion`.
                  'mirror' is the default.
-
-        Raises:
-            ValueError: if 'sync_mode' is passed an invalid value.
         """
-        if sync_mode != 'mirror' and sync_mode != 'additive':
-            msg = _("'sync_mode' must either be 'mirror' or 'additive' not '{sync_mode}'")
-            raise ValueError(msg.format(sync_mode=sync_mode))
+        assert(sync_mode == 'mirror' or sync_mode == 'additive')
```

Comment:
I don't think we should use assert here. Instead we should raise a `ValueError` which is already there. I think this file should be left unedited.